### PR TITLE
[MRG+1] Deepcopy of info in array constructors.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/3682

We already do it for ``EpochsArray``.
Also closes https://github.com/mne-tools/mne-python/issues/3683

### CODE REVIEW

**Comment by jona-sassenhagen** [comment]
Looks good to me.

File: mne/evoked.py [hunk 1]
```diff
@@ -702,7 +702,7 @@ def __init__(self, data, info, tmin, comment='', nave=1, kind='average',
         self.last = self.first + np.shape(data)[-1] - 1
         self.times = np.arange(self.first, self.last + 1,
                                dtype=np.float) / info['sfreq']
-        self.info = info
+        self.info = deepcopy(info)  # do not modify original info
```
Comment (Eric89GXL) [inline mne/evoked.py]: You can just use `info.copy()`

File: mne/io/array/array.py [hunk 2]
```diff
@@ -54,6 +55,7 @@ def __init__(self, data, info, first_samp=0, verbose=None):  # noqa: D102
         assert len(info['ch_names']) == info['nchan']
         if info.get('buffer_size_sec', None) is None:
             info['buffer_size_sec'] = 1.  # reasonable default
+        info = copy.deepcopy(info)  # do not modify original info
```
Comment (Eric89GXL) [inline mne/io/array/array.py]: same here

File: mne/evoked.py [hunk 3]
```diff
@@ -681,7 +681,7 @@ class EvokedArray(Evoked):
     """
 
     @verbose
-    def __init__(self, data, info, tmin, comment='', nave=1, kind='average',
+    def __init__(self, data, info, tmin=0., comment='', nave=1, kind='average',
```
Comment (Eric89GXL) [inline mne/evoked.py]: in `tests/test_evoked.py` can you modify a test to make use of this new default?

**Comment by kingjr** [comment]
Thanks @jaeilepp !

File: mne/io/fiff/tests/test_raw_fiff.py [hunk 4]
```diff
@@ -1141,7 +1142,7 @@ def test_add_channels():
     assert_raises(ValueError, raw_meg.copy().add_channels, [raw_arr])
     raw_meg.copy().add_channels([raw_arr], force_update_info=True)
     # Make sure that values didn't get overwritten
-    assert_true(raw_arr.info['dev_head_t'] is orig_head_t)
+    _is_equal_dict([raw_arr.info['dev_head_t'], orig_head_t])
```
Comment (Eric89GXL) [inline mne/io/fiff/tests/test_raw_fiff.py]: don't you need to `assert_true(_is_equal_dict(...))`? If not, then that is a misnomer and should be renamed `_assert_equal_dict`

Comment (jaeilepp) [inline mne/io/fiff/tests/test_raw_fiff.py]: Yeah you're right. Good catch.

**Comment by Eric89GXL** [comment]
Ready to go from your end @jaeilepp ?

**Comment by kingjr** [comment]
travis isn't happy

**Comment by Eric89GXL** [comment]
@jaeilepp I assume this is MRG from your end? LGTM so I changed the title to MRG+1

**Comment by jaeilepp** [comment]
Yeah, ready from my end. Travis failed earlier for reasons I don't understand. Restarting made it pass.

**Comment by agramfort** [comment]
thx @jaeilepp

**Pull request merged**